resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -606,21 +606,19 @@ private boolean doResolveCached(String hostname,
 
         InetAddress address = null;
         Throwable cause = null;
-        synchronized (cachedEntries) {
-            final int numEntries = cachedEntries.size();
-            assert numEntries > 0;
+        final int numEntries = cachedEntries.size();
```

Comment:
move this into the conditional block where it is actually used. If the assertion is not there `get(0)` will throw, and we already check for `isEmpty()` above.